### DISCUSSION TITLE
[Feature]: Recommended way to handle dynamic nested fields in forms

### DISCUSSION THREAD

**Discussion by Eventect** [remix-run/remix#2748] [Proposals]
### What is the new or updated feature that you are suggesting?

Is there a recommended way to handle dynamic nested fields in forms, similar to what the (fields_for helper)[https://apidock.com/rails/v6.1.3.1/ActionView/Helpers/FormHelper/fields_for] does in Rails or (express.bodyParser middleware)[http://expressjs.com/en/resources/middleware/body-parser.html] in express?

### Why should this feature be included?

There is no clear way, without writing a custom parser, to implement a nested form working without javascript, which as far as I understand, is a pivotal principle in the <Form> implementation in remix.

**Comment by sbernheim4** [comment]
I have a couple ideas. 

## First Idea

If you're aiming for a no JS based solution:

Split your forms up wherever the conditional boundaries lie. 

In your form action handler, capture each submission and return some data indicating if the next part of the form should be shown given the selections made. 

In the component, use the action hook to retrieve that same data and conditionally render the next form as appropriate. 

## My other idea 

If you want to do it via JS on the client, store responses to the questions in a useState value and conditionally render other forms based on the state variables.
